src/main/java/com/jcabi/beanstalk/maven/plugin/AbstractBeanstalkMojo.java

```diff
@@ -259,6 +269,80 @@ protected boolean isGreen(final Environment env) {
     }
 
     /**
+     * Validate given text in YAML format.
+     *
+     * @param text YAML text
+     * @return Boolean true if valid else false
+     */
+    protected boolean validYaml(final String text) {
+        boolean valid = true;
+        try {
+            new Yaml().load(text);
+        } catch (final YAMLException exception) {
+            valid = false;
+        }
+        return valid;
+    }
+
+    /**
+     * Validate the war file.
+     *
+     * @param zip Zip file
+     * @throws org.apache.maven.plugin.MojoFailureException Thrown, if
+     *  validation fails.
+     */
+    protected void validate(final ZipFile zip) throws MojoFailureException {
+        final ZipEntry ebextdir = zip.getEntry(".ebextensions");
+        if (ebextdir == null) {
+            throw new MojoFailureException(
+                ".ebextensions directory does not exist in the WAR file"
+            );
+        }
+        final Enumeration<? extends ZipEntry> entries = zip.entries();
+        int files = 0;
+        while (entries.hasMoreElements()) {
+            final ZipEntry entry = entries.nextElement();
+            if (entry.getName().startsWith(".ebextensions/")
+                && !entry.isDirectory()) {
+                files += 1;
+                final String text = this.readFile(zip, entry);
+                if (this.validYaml(text)) {
+                    throw new MojoFailureException(
+                        Joiner.on("").join(
+                            "File '",
+                            entry.getName(),
+                            "' in .ebextensions is not a valid YAML"
+                        )
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Reads text from a ZIP file.
+     * @param warfile ZIP file, which contains entry.
+     * @param entry ZIP entry (compressed file) to read from.
+     * @return Text content of entry.
+     */
+    private String readFile(final ZipFile warfile, final ZipEntry entry) {
+        String text = null;
+        InputStream inputStream = null;
+        InputStreamReader reader = null;
+        try {
+            inputStream = warfile.getInputStream(entry);
+            reader = new InputStreamReader(inputStream);
+            text = CharStreams.toString(reader);
+        } catch (final IOException exception) {
+            Logger.error(this, exception.getMessage());
```

Comment:
@dmzaytsev catch and thrown as MojoExecutionException instead. If we throw it as IOE, we'll still need to convert it later to MojoExcecution in `execute()` method.